src/lib/storage/buffer/page_id.hpp

```diff
@@ -1,62 +1,109 @@
+
+// Get the minimum PageSizeType. KiB16 on Mac OS and KiB4 on Linux
 constexpr PageSizeType MIN_PAGE_SIZE_TYPE = magic_enum::enum_value<PageSizeType>(0);
-constexpr PageSizeType MAX_PAGE_SIZE_TYPE = magic_enum::enum_value<PageSizeType>(NUM_PAGE_SIZE_TYPES - 1);
-constexpr size_t PAGE_SIZE_TYPE_BITS = boost::static_log2<NUM_PAGE_SIZE_TYPES>::value + 1;
+
+// Get the maximum PageSizeType. MiB2 on Mac OS and Linux
+constexpr PageSizeType MAX_PAGE_SIZE_TYPE = magic_enum::enum_value<PageSizeType>(PAGE_SIZE_TYPES_COUNT - 1);
+
+// Get the number of bits required to store a PageSizeType
+// TODO(nikriek): Replace with std::bit_width once we get rid of gcc9
+constexpr uint64_t PAGE_SIZE_TYPE_BITS =
+    std::numeric_limits<uint64_t>::digits - std::countl_zero(PAGE_SIZE_TYPES_COUNT);
 
 /**
- * PageIDs are used for addressing pages. They consist of a valid flag, a PageSizeType and an index. A Page ID can be unambiguously
- * converted into a virtual address and back. If the valid flag is set, the page is stored in the buffer pool. Otherwise, the page
+ * PageIDs are used for addressing pages. They consist of a validity flag, a PageSizeType, and an index. A PageID can be unambiguously
+ * converted into a virtual address and back. The valid flag indicates that the page is stored in the buffer pool. Otherwise, the page
  * does not exist or the virtual memory address is outside of the buffer pool.
+ * 
+ * For the implementation, we use C++ bitfields to compress multiple fields into a single 64-bit values without manual bit shifting. 
+ * The valid flag is stored in the most significant bit. The PageSizeType is stored in the next PAGE_SIZE_TYPE_BITS bits. The index 
+ * is stored in the remaining bits.
 */
 struct PageID {
   using PageIDType = uint64_t;
 
-  PageIDType _valid : 1;
-  PageIDType _size_type : PAGE_SIZE_TYPE_BITS;
-  PageIDType index : (sizeof(PageIDType) * CHAR_BIT - PAGE_SIZE_TYPE_BITS - 1);
+  constexpr explicit PageID(const PageSizeType size_type, const uint64_t index, bool valid = true)
+      : _valid(valid), _size_type(static_cast<PageIDType>(size_type)), _index(index) {}
 
+  // Get the PageSizeType for the page
   PageSizeType size_type() const {
-    return magic_enum::enum_value<PageSizeType>(this->_size_type);
+    return magic_enum::enum_value<PageSizeType>(_size_type);
   }
 
-  size_t num_bytes() const {
+  // Get the number of bytes for the page
+  uint64_t byte_count() const {
     return bytes_for_size_type(size_type());
   }
 
+  // Get the index of the page of a size type
+  uint64_t index() const {
+    return _index;
+  }
+
+  // Returns if the PageID is valid or not in the buffer pool
   bool valid() const {
     return _valid;
   }
 
-  PageID() = default;
+  bool operator==(const PageID& other) const {
+    return (_valid == other._valid) &&
+           (!_valid || (_valid && _size_type == other._size_type && _index == other._index));
```

Comment:
```suggestion
           (!_valid || (_size_type == other._size_type && _index == other._index));
```

We don't need to check `_valid` again since the the right operand of `||` is only evaluated if `_valid` is true.